vectorsearch/resource_vector_search_endpoint.go

```diff
@@ -0,0 +1,85 @@
+package vectorsearch
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/databricks/databricks-sdk-go/service/vectorsearch"
+)
+
+const DefaultProvisionTimeout = 45 * time.Minute
+
+func ResourceVectorSearchEndpoint() *schema.Resource {
+	s := common.StructToSchema(
+		vectorsearch.EndpointInfo{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(s, "name").SetRequired()
+			s["name"].ForceNew = true
+			s["endpoint_type"].ForceNew = true
+			common.CustomizeSchemaPath(s, "endpoint_type").SetRequired()
+			delete(s, "id")
+			common.CustomizeSchemaPath(s, "creator").SetReadOnly()
+			common.CustomizeSchemaPath(s, "creation_timestamp").SetReadOnly()
+			common.CustomizeSchemaPath(s, "last_updated_timestamp").SetReadOnly()
+			common.CustomizeSchemaPath(s, "last_updated_user").SetReadOnly()
+			common.CustomizeSchemaPath(s, "endpoint_status").SetReadOnly()
+			common.CustomizeSchemaPath(s, "num_indexes").SetReadOnly()
+			common.CustomizeSchemaPath(s).AddNewField("endpoint_id", &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			})
+
+			return s
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var req vectorsearch.CreateEndpoint
+			common.DataToStructPointer(d, s, &req)
+			endpoint, err := w.VectorSearchEndpoints.CreateEndpointAndWait(ctx, req,
+				retries.Timeout[vectorsearch.EndpointInfo](d.Timeout(schema.TimeoutCreate)))
```

Comment:
We're deprecating the `*AndWait` methods. Can we use `GetWithTimeout` instead?
```suggestion
			endpoint, err := w.VectorSearchEndpoints.CreateEndpoint(ctx, req).GetWithTimeout(d.Timeout(schema.TimeoutCreate))
```

Reply:
Thank you - somehow it wasn't visible in the code completion